### PR TITLE
Various tidy ups

### PR DESCRIPTION
**Pull request by ltratt**
A number of tidy-ups I noticed while understanding bits of the JIT IR.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 1]
```diff
@@ -17,13 +17,10 @@
 //!  * `Inst`: "instruction"
 //!  * `Ty`: "type"
 //!
-//! To get human-readable representations of IR data structures, most of the IR constructs either:
-//!  * implement [std::fmt::Display] and thus can be stringified with (e.g.) `format!`, or
-//!  * if extra information is needed, have a method called `display()` which takes extra arguments
-//!    and returns something which does implement [std::fmt::Display].
-//!
-//! In the human readable IR, all names should be lowercase. Try to keep them short. Underscores
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir.rs:25]: I guess you thought this bit was obvious?

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:25]: Yes.

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 2]
```diff
@@ -687,23 +687,13 @@ impl FuncTy {
     }
 }
 
-/// A structure's type.
```
[the remark this answers is not in the document]
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:692]: Maybe. But it's dead code, so arguably shouldn't have been in here in the first place. We can easily fish it out of git history if/when we need it.

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 3]
```diff
@@ -22,10 +22,6 @@
 //!  2. or, when they need extra information, they expose a `display()` method, which returns an
 //!     object which implements [std::fmt::Display].
 
-// For now, don't swamp others working in other areas of the system.
-// FIXME: eventually delete.
-#![allow(dead_code)]
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir.rs:30]: OK, now I see why you are being to aggressive with removing dead, but potentially needed later code :)

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:30]: I'm transparent :)

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 4]
```diff
@@ -150,20 +151,12 @@ impl Module {
     /// # Panics
     ///
     /// Panics if the address cannot be located.
+    #[cfg(not(test))]
     pub(crate) fn globalvar_ptr(&self, idx: GlobalDeclIdx) -> *const () {
         let decl = self.global_decl(idx);
-        #[cfg(not(test))]
```
[the remark this answers is not in the document]
Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir.rs:158]: Ok, we can bring it back later.

**Pull request merged**